c++/src/kj/common.h

```diff
@@ -1954,6 +1954,13 @@ class ArrayPtr: public DisallowConstCopyIfNotConst<T> {
     // libc++ std::fill doesn't have memset specialization either.
   }
 
+  inline void copyFrom(kj::ArrayPtr<T> other) {
+    // Copy data from the other array pointer.
+    // Arrays have to be of the same size and memory area MUST NOT overlap.
```

Comment:
`memcpy` has non-overlapping requirement too. I could implement overlapping copy if you think it is important.

Reply:
I don't think we have any cases currently where we have an overlapping copy so I think this is fine. I do wonder if it wouldn't be worthwhile to at least have a `KJ_DASSERT` in here ensuring `other` is not overlapping?